### ISSUE TITLE
Inconsistent handling of Key arguments when provided at configuration time (flag is always false)

### ISSUE THREAD

**Issue by hannesm** [mirage/mirage#1074]
Here's a slightly modified version of `hello-key` from mirage-skeleton:

config.ml:
```OCaml
open Mirage

let hello =
  let doc = Key.Arg.info ~doc:"Whether to say hello." ["hello"] in
  Key.(create "hello" Arg.(opt bool false doc))

let flag =
  let doc = Key.Arg.info ~doc:"Whether to really say hello." ["flag"] in
  Key.(create "flag" Arg.(flag doc))

let main =
  foreign
    ~keys:[Key.abstract hello; Key.abstract flag]
    ~packages:[package "duration"]
    "Unikernel.Hello" (time @-> job)

let () =
  register "hello" [main $ default_time]
```

unikernel.ml:
```OCaml
open Lwt.Infix

module Hello (Time : Mirage_time.S) = struct

  let start () =
    let hello = "hello world" in
    let rec loop = function
      | 0 -> Lwt.return_unit
      | n ->
        Logs.info (fun f -> f "hello %B and flag %B" (Key_gen.hello ()) (Key_gen.flag ()));
        if Key_gen.hello () && Key_gen.flag () then
          Logs.info (fun f -> f "%s" hello);
        Time.sleep_ns (Duration.of_sec 1) >>= fun () ->
        loop (n-1)
    in
    loop 4

end
```

Now, if we run `mirage configure && make && ./main.native`, both `hello` and `flag` will show as `false`. We can run `./main.native --hello true --flag` to see that both will be `true` (this works fine).

If we would like to embed at configuration time some default value (i.e. `true`), we can run `mirage configure --hello true --flag && make && ./main.native`, but while this prints `hello` as `true`, `flag` is `false`.

TL;DR: It looks like any `flag` is configuration-time always set to `false`, and there's no way to set it to true. In addition, I wonder whether the `flag` vs `opt bool` complexity is intended & necessary.

//cc @linse @mato (encountered this issue while working on muen which does hardly allow boot parameters, best to select at configuration time).

**Comment by mato** [comment]
> In addition, I wonder whether the flag vs opt bool complexity is intended & necessary.

I'll note that `opt bool` does not have this problem, as I discovered when initially trying to reproduce this with hello-key and failing.

**Comment by hannesm** [comment]
Since Mirage 4.5.0, we have a separate story and no "both stage" keywords. This also means that there's no serialisation happening anymore, and this issue is void.
